### PR TITLE
Allow users to submit comments

### PR DESCRIPTION
**Pull request by ganewman**
Finish step 4 of this week's walkthrough. Add form on homepage to allow users to submit comments, and modify servlet to support temporarily storing and then displaying these comments.

The code is written with the intention of displaying the comments section on all pages of the site, not just the homepage. I haven't yet done it this at this stage because I think it will make the code harder to modify in the near future.

### CODE REVIEW

File: portfolio/src/main/webapp/index.html [hunk 1]
```diff
@@ -29,6 +29,12 @@ <h1>Gabi Newman's Portfolio</h1>
       <div id="fact-container"></div>
       <div id="comments">
         <h3>Comments</h3>
+        <form action="/data" method="POST">
+        <textarea name="comment-input">Your comment here</textarea>
```
Comment (kellyktan) [inline portfolio/src/main/webapp/index.html:33]: Is there a way to clear this when a user clicks in to edit it? Right now you have to manually delete it if you don't want it as part of your comment

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -41,6 +41,15 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.getWriter().println(json);
   }
 
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```
Comment (kellyktan) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:44]: Does this fulfill the tODO at the top of the class?

File: portfolio/src/main/webapp/index.html (hunk 1 above)
Comment (ganewman) [reply portfolio/src/main/webapp/index.html]: Suparna also pointed this out. I agree it's super annoying. I think it should be good now.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (ganewman) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:44]: I'm not sure... I was going to wait until I had finished all of the steps of the walkthrough to get rid of it. I have one left to go.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -41,6 +41,15 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+    String userComment = request.getParameter("comment-input");
+    // Gets page the user sent the request from using a hidden form
```
Comment (suparnabarua) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:46]: "Uses a hidden form to get the page the user sent the request from."

Comment (ganewman) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Oops, grammar.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (kellyktan) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:44]: sounds good to me (SGTM)

**Pull request merged**